### PR TITLE
Add more ansible calls

### PR DESCRIPTION
**Pull request by ycao56**
https://github.com/open-cluster-management/backlog/issues/11114
1. moved ansible related code to its own file
2. added call `ansibleCredentials` to list all the root ansible credentials
3. added call `copyAnsibleSecret` to return copied ansible secret name

image

image

### CODE REVIEW

File: src/v1/models/ansible.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    if (namespace !== targetNamespace) {
+      // check if credentianl has been already created
+      const secret = await this.kubeConnector.get(
+        `/api/v1/namespaces/${targetNamespace}/secrets?labelSelector=cluster.open-cluster-management.io/copiedFromSecretName=${name},cluster.open-cluster-management.io/copiedFromNamespace=${namespace}`,
+      );
+      if (!secret.items) {
+        throw new Error(`Failed to retrieve copied secrets from ${targetNamespace}`);
+      } else {
+        if (secret.items.length === 0) {
+          // no secret need to copy and return
```
Comment (ChunxiAlexLuo) [inline src/v1/models/ansible.js:56]: // no target secret in 'targetNamespace', need to copy from rootSecret and return secret name

**Review by ChunxiAlexLuo — APPROVED** [review]
ltgm

File: src/v1/models/ansible.js (hunk 1 above)
Comment (ycao56) [reply src/v1/models/ansible.js]: done

**Pull request merged**